### ISSUE TITLE
World Orientation Adjustment (horizon adjust)

### ISSUE THREAD

**Issue by kulzae** [Yellow-Dog-Man/Resonite-Issues#828]
### Is your feature request related to a problem? Please describe.

Currently it can be difficult to use resonite while reclining or laying in bed due to your avatar always looking up (as you are not looking straight ahead when laying down) It would be useful to be able to adjust the orientation of the world in relation to the headsets orientation to allow you to be looking forward when your actual orientation is looking up or angled. This has been previously referred to as "bed mode" and is known as horizon adjust in other platforms.

### Describe the solution you'd like

Here are some images from the perspective of someone looking at the vr user to better describe the issue
![Standing Before Adjustment](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/40366296/219e43ca-d514-4f87-a834-ccd6c9b24f74)
This image shows the current mode resonite operates in. with the user standing and the RL ground aligned with the ground plane of resonite

![Laying After Adjustment](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/40366296/2479a784-faf6-4aae-a315-00112bc6c2fc)

This would be after adjustment. the RL ground plane remaining the same but the orientation of the user changing and orientation of resonite and the view the user sees changing to go with it. this would be something that is set once and re oriented or reset as needed

### Describe alternatives you've considered

Currently i use three axis noclip or fly to reorient my view but it requires locomotion changes, my avatar to be floating, and my legs to be behind me as the game is still assuming im looking upward. This feature would also make it easier for those who cannot sit or stand upright to use the platform naturally without needing specific avatar setups or changes to compensate for the angled upward view

**Comment by Frooxius** [comment]
This is something I wanted to implement at some point. It's not that complex, so it shouldn't take long too add.

**Comment by shadowpanther** [comment]
Could it be an extension of the seated mode?

- When seated mode is toggled on, the current position of the head is saved as a standing straight position.
- When laid down mode is toggled on, the current position _and rotation_ of the head is saved as a standing straight position/rotation.

**Comment by shadowpanther** [comment]
> I would want them to be separate options/toggles, as there are use-cases/reasons why a user might want to use one but not the other, @shadowpanther. While there is some synergy in their use- they do solve different problems.

I meant for them to be separate options, but the underlying mechanic (recentering and rotating the playspace around the user's head) might be the same. Sorry if my description looked like I suggested for them to be one option.

**Comment by epicEaston197** [comment]
This is a really useful accessibility feature and now that the settings menu is completed this could be unblocked and could be considered

**Comment by Frooxius** [comment]
It already is. You can see above that it's been moved to things to do.

**Comment by GearBell** [comment]
What if in the Settings menu you can set what angle you want to offset, so when you press the button its to your own liking.
I hope this is a real thing, Im stuck laying down due to injury for the next few weeks and am trying to figure out how to juryrig it for now.
A set offset settings you input yourself means Froox wouldnt need to figure out what everyone wants, cause heaven knows thats a mess and a half. It would mean it just unlocks that ability for us to mess with.

**Comment by ShaunDreclin** [comment]
Is this still not available? It's a pretty basic accessibility feature, I think it should be more of a priority to implement.

I had a back injury that left me bed-bound for the better part of a year and horizon adjust on VRC was vital for my ability to continue using VR as a social outlet.

**Comment by Frooxius** [comment]
The issue is still open and it's in the "Todo" list on the right side panel, meaning it has not been implemented yet. Typically the issue is closed when a feature is implemented.

There are many features to work on, so not everything will happen at once unfortunately - we are limited on our time.

If you'd like to give it more priority, please give the issue an upvote.

I've added the accessibility flag too, which makes us more likely to prioritize it.

**Comment by ShaunDreclin** [comment]
> I've added the accessibility flag too, which makes us more likely to prioritize it.

Appreciated! Gave it an upvote
